### PR TITLE
Refactor/clean up class and test files

### PR DESCRIPTION
**Pull request by lschnell8**
PR Template:

Is this a fix or a feature?
- [ ] Bug fix
- [ ] Feature
- [x] Refactor
 
Describe the feature - fix - refactor:
Refactor class and test files 

How was this tested?
- [ ] open index.html
- [ ] localhost
- [x] terminal

### CODE REVIEW

File: src/classes/RecipeBox.js [hunk 1]
```diff
@@ -1,51 +1,43 @@
...
-    ingredientsData.forEach((ingredient) => {
-      if (ingredient.name === theUserInput) {
-        matchingIngredientId = ingredient.id
-      }
-    })
-    this.recipesCollection.forEach((recipe) => {
-      recipe.ingredients.forEach((ingredient) => {
-        if (ingredient.id === matchingIngredientId) {
-          this.recipeMatch.push(recipe)
+  findIngredientById(theUserInput) {
```
Comment (Alynn022) [inline src/classes/RecipeBox.js:30]: So much cleaner! This!

File: src/classes/User.js [hunk 2]
```diff
@@ -56,20 +51,20 @@ class User {
...
-    ingredientsData.forEach((ingredient) => {
-      if (ingredient.name === userInput) {
-        matchingIngredientId = ingredient.id
-      }
-    })
-    this.favoriteRecipes.forEach((recipe) => {
-      recipe.ingredients.forEach((ingredient) => {
-        if (ingredient.id === matchingIngredientId) {
-          this.recipeMatch.push(recipe)
+    return this.favoriteRecipes.reduce((acc, recipe) => {
```
Comment (Alynn022) [inline src/classes/User.js:54]: I like that we got rid of the empty array

File: src/classes/Recipe.js [hunk 3]
```diff
@@ -13,34 +13,35 @@ class Recipe {
...
+    })
+  }
   
   calculateRecipeCost() {
     let totalCost;
-       const total = this.ingredients.reduce((acc, recipeIngredient) => {
-         let foundID = ingredientsData.find(elem => {return elem.id === recipeIngredient.id});
-         totalCost = (recipeIngredient.quantity.amount * foundID.estimatedCostInCents) / 100;
-         acc += totalCost;
+    const total = this.ingredients.reduce((acc, recipeIngredient) => {
```
Comment (StevieAmb) [inline src/classes/Recipe.js:34]: Excellent refactor!

**Review by Alynn022 — APPROVED** [review]

File: src/classes/User.js (hunk 2 above)
Comment (StevieAmb) [inline src/classes/User.js:54]: This is an incredible display of chaining iterator methods.

**Review by StevieAmb — APPROVED** [review]

**Pull request merged**